### ISSUE TITLE
Rounding Error in Mask RCNN training script

### ISSUE THREAD

**Issue by twankim** [dmlc/gluon-cv#1447]
https://github.com/dmlc/gluon-cv/blob/cf767a1297d67caf41577508aacc7cbc24ae8f14/scripts/instance/mask_rcnn/train_mask_rcnn.py#L430-L438

This can cause rounding error, which results in mismatch between ground truth and predicted masks' image size.
mAP scores on pycocotools may degrade.
Same for `validate` function in evaluation script.

I'd recommend directly provide the original image's shape in `MaskRCNNDefaultValTransform` by replacing this part:
https://github.com/dmlc/gluon-cv/blob/cf767a1297d67caf41577508aacc7cbc24ae8f14/gluoncv/data/transforms/presets/rcnn.py#L458-L468

into :
```
def __call__(self, src, label, mask):
     """Apply transform to validation image/label."""
    # resize shorter side but keep in max_size
    h, w, _ = src.shape
    img = timage.resize_short_within(src, self._short, self._max_size, interp=1)

    img = mx.nd.image.to_tensor(img)
    img = mx.nd.image.normalize(img, mean=self._mean, std=self._std)
    return img, mx.nd.array([img.shape[-2], img.shape[-1], h, w])
```

And change `validate` functions as
```
                im_height, im_width, im_orig_height, im_orig_width = det_info
                im_scale = float(im_height) / im_orig_height
                .
                .
                .
                # fill full mask
                full_masks = gdata.transforms.mask.fill(det_mask, det_bbox, (int(im_orig_width), int(im_orig_height)))
```
